src/back/glsl/mod.rs

```diff
@@ -453,6 +490,10 @@ impl<'a, W: Write> Writer<'a, W> {
         // This are always ordered because of the IR is structured in a way that you can't make a
         // struct without adding all of it's members first
         for (handle, ty) in self.module.types.iter() {
+            if self.unused_unsized_types.contains(&handle) {
+                continue;
+            }
+
             if let TypeInner::Struct { ref members, .. } = ty.inner {
                 // No needed to write a struct that also should be written as a global variable
                 let is_global_struct = self
```

Comment:
oh, look! here we are already doing the scan of globals, finding the variable corresponding to this type.
This code ensures that we aren't writing down any structs that have globals using them.
I think this code needs to be fixed in a number of ways:
  - support the case where there is a global using the struct, and also there are local users. Currently, the logic allows them exclusive.
  - skip globals that aren't used by the entry point

But we don't have to do this in your PR, we can follow-up.
The important takeaway here is that this code would already skip any structs that are used by globals, so you don't need to add any extra checks here.

Reply:
Oh, interesting! I'll take that check out.